src/cpu.rs

```diff
@@ -0,0 +1,557 @@
+use crate::flags::ConditionFlags;
+use crate::memory::Memory;
+use crate::opcode::{Opcode, Trap};
+use std::io::{self, Read, Write};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CPUErrors {
+    #[error("Error obtaining the register value")]
```

Comment:
This Error can be improved:
- By convention the type names are in singular instead of plural (it should be `CPUError`).
- It's useful to add as much information as possible to the error types. That will help the one handling the error to better understand what is happening. 

In this case you can add the register name to the `Register` variant.
The instruction name to `Execute`